From a97814436ee5c95397ab9b5a1bebecb305748da7 Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Fri, 10 Apr 2026 11:47:39 -0300
Subject: [PATCH] Customize Debian build files for STX

Change the Debian build files to generate qemu Debian packages as
they are needed for StarlingX qemu use case.
Building the upstream qemu was taking ~30 minutes and generating
an output directory with 30 packages and ~1.1G in size. Removing
the unnecessary packages resulted in a build that takes ~15min and
in an output directory with only 10 needed packages and ~320M.

This removes UI/Graphical modules and focuses on building
qemu-system-x86 only. Other qemu-system-* packages for different
architectures are not required, neither the build dependencies for
cross-compiling those packages. Also, this reduces the configure
and build targets to the ones that really matter for StarlingX.

Removed packages:
* qemu-system-gui
* qemu-system-modules-spice (split from gui in 8.1 [1])
* qemu-system-modules-opengl (split from common in 8.1 [2])
* qemu-system-[misc|arm|mips|ppc|sparc]
* qemu-system-riscv (split from misc in 9.1 [3])
* qemu-system-s390x (split from misc in 9.1 [3])
* qemu-system-xen and ubuntu transitional qemu-system-x86-xen
* qemu-user
* qemu-user-static
* qemu-user-binfmt

Removed non-x86 firmware builds from debian/rules:
* openbios, skiboot, SLOF, vof
* s390x-fw, hppa-fw
* u-boot-e500, u-boot-sam460
* palcode-clipper, vbootrom

Removed xen build (QEMU_XEN variable, configure/build/install-xen).

Cleaned up qemu-system metapackage dependencies to only keep
qemu-system-x86.

Removed libvirt-daemon Breaks constraint from qemu-system-common
as libvirt in Trixie is already above 7.2.

Note: since qemu 9.0.2 [4], the former debian/control file has been
split into debian/control-in and debian/control.mk. Build target
lists (system-packages, user-targets) are now defined in control.mk.

This patch consolidates the following previously separate patches
into a single one, as they all serve the same purpose of customizing
the build for STX x86-only:
* 0002-remove-xen-build-for-STX.patch
* 0003-rules-remove-vof.patch
* 0005-remove-dependency-on-qemu-system-ARCH.patch
* 0007-workaround-for-libvirt-breaks.patch
The following patch was dropped as it is no longer needed in 10.0.8
(the build system now handles it conditionally):
* 0004-add-condition-for-dh_installdocs.patch

Packaging follow-up fixes (consequences of the above removals):

1. debian/not-installed: Add files that are still built but no longer
   assigned to any package after the removal of qemu-system-gui and
   qemu-user:
   - usr/share/locale/* (locale files from qemu-system-gui)
   - usr/bin/qemu-system-* (system binaries installed via d/rules)
   - usr/share/doc/qemu/user/* (user docs from qemu-user)
   - 18 orphaned .so modules: audio backends (dbus, jack, pa,
     pipewire, sdl, spice), UI modules (dbus, egl-headless, gtk,
     opengl, sdl, spice-app, spice-core), display modules (qxl,
     virtio-gpu-gl, virtio-gpu-pci-gl, virtio-vga-gl), and
     chardev-spice.

2. debian/qemu-system-common.install:
   - Add vhost-user GPU files (50-qemu-gpu.json, vhost-user-gpu)
     previously owned by qemu-system-gui.
   - Remove chardev-baum.so since brlapi is disabled
     (--disable-brlapi) and the module is never built.

3. debian/rules:
   - Remove -Nqemu-user from dh_shlibdeps since qemu-user package
     no longer exists.

[1] https://salsa.debian.org/qemu-team/qemu/-/commit/0dc095811b
[2] https://salsa.debian.org/qemu-team/qemu/-/commit/4a0400dd2e
[3] https://salsa.debian.org/qemu-team/qemu/-/commit/6ca9c8f608
[4] https://salsa.debian.org/qemu-team/qemu/-/commit/b343f7840f

Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
[Rebased original changes to the qemu 7.2 version]
Signed-off-by: david.liu <david.liu@windriver.com>
[Aligned with 7.2, updated configure with --target-list]
Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
[Co-author of consolidated patches listed above]
Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
[Rebased to qemu 10.0.8 for Trixie, added packaging follow-up fixes]
Signed-off-by: cbohlhal <Caique.BohlhalterdeSouza@windriver.com>
---
 debian/control-in                 | 385 +-----------------------------
 debian/control.mk                 |  33 +--
 debian/not-installed              |  21 ++
 debian/qemu-system-common.install |   3 +-
 debian/rules                      | 186 ++-------------
 5 files changed, 55 insertions(+), 573 deletions(-)

diff --git a/debian/control-in b/debian/control-in
index 11dadaf9cf..34a95192b3 100644
--- a/debian/control-in
+++ b/debian/control-in
@@ -32,18 +32,18 @@ Build-Depends-Arch:
 # libsndio is in universe in ubuntu
 # --disable-sndio
 # --audio-drv-list=pipewire,pa,alsa,jack,oss,sdl	:system-arch-linux:
- libjack-dev    [:system-arch-linux:],
- libpulse-dev   [:system-arch:],
- libasound2-dev [:system-arch-linux:],
- libpipewire-0.3-dev [:system-arch-linux:],
+# libjack-dev    [:system-arch-linux:],
+# libpulse-dev   [:system-arch:],
+# libasound2-dev [:system-arch-linux:],
+# libpipewire-0.3-dev [:system-arch-linux:],
 # for virtfs (now in libc6)
 # --enable-attr
 # --enable-bpf	 :system-arch-linux:
  libbpf-dev	[:system-arch-linux:],
 # --enable-blkio	:system-arch-linux-64:
  libblkio-dev	       [:system-arch-linux-64:],
-# --enable-brlapi	:system-arch:
- libbrlapi-dev         [:system-arch:],
+# --disable-brlapi
+# libbrlapi-dev         [:system-arch:],
 # --enable-virtfs	:system-arch-linux:
 # needed for virtfs
 # --enable-cap-ng	:system-arch-linux:
@@ -107,7 +107,8 @@ Build-Depends-Arch:
 # vde is debian-only since ubuntu/vde2 is in universe
 :debian:# --enable-vde	 :system-arch:
 :debian: libvdeplug-dev	[:system-arch:],
- libxen-dev [linux-amd64],
+# --disable-xen
+# libxen-dev [linux-amd64],
 # --enable-nettle	 :system-arch:
  nettle-dev		[:system-arch:],
 # other optional features we enable
@@ -156,21 +157,20 @@ Build-Depends-Indep:
  gcc-i686-linux-gnu,
 # pc-bios/*.dts => *.dtb (PPC firmware)
  device-tree-compiler,
- gcc-s390x-linux-gnu,
+# gcc-s390x-linux-gnu,
 # qemu-palcode/palcode-clipper
  gcc-alpha-linux-gnu,
 # u-boot code
  gcc-powerpc-linux-gnu,
 # skiboot firmware, openbios
- gcc-powerpc64-linux-gnu,
+# gcc-powerpc64-linux-gnu,
 # skiboot includes <openssl/something.h>
  libssl-dev,
 # openbios
- gcc-sparc64-linux-gnu, fcode-utils, xsltproc,
+# gcc-sparc64-linux-gnu, fcode-utils, xsltproc,
 # hppa-firmware (32 and 64 bit)
- gcc-hppa-linux-gnu, gcc-hppa64-linux-gnu,
- gcc-riscv64-linux-gnu,
-# vbootrom/npcm7xx
+# gcc-hppa-linux-gnu, gcc-hppa64-linux-gnu,
+# gcc-riscv64-linux-gnu,
  gcc-arm-linux-gnueabi, libc6-dev-armel-cross,
 # vbootrom/npcm8xx
  gcc-aarch64-linux-gnu, libc6-dev-arm64-cross,
@@ -199,14 +199,7 @@ Architecture: :system-arch:
 Multi-Arch: foreign
 Build-Profiles: <!pkg.qemu.omit-system>
 Depends: ${misc:Depends},
- qemu-system-arm,
- qemu-system-mips,
- qemu-system-ppc,
- qemu-system-riscv,
- qemu-system-s390x,
- qemu-system-sparc,
  qemu-system-x86,
- qemu-system-misc
 Description: QEMU full system emulation binaries
  QEMU is a fast processor emulator: currently the package supports Alpha, ARM,
  CRIS, i386, LoongArch, M68k (ColdFire), MicroBlaze, MIPS, PowerPC, RISC-V,
@@ -298,7 +291,6 @@ Package: qemu-system-common
 Architecture: :system-arch:
 Multi-Arch: no
 Build-Profiles: <!pkg.qemu.omit-system>
-Breaks: libvirt-daemon (<< 7.2.0-1)
 Depends: ${misc:Depends}, ${shlibs:Depends},
 # to fix wrong acl for newly created device node on ubuntu:
 :ubuntu: acl
@@ -312,279 +304,16 @@ Description: QEMU full system emulation binaries (common files)
  This package provides common files needed for target-specific
  full system emulation (qemu-system-*) packages.
 
-Package: qemu-system-gui
-Architecture: :system-arch:
-#XXX M-A: same does not really work for now due to /usr/lib/qemu/vhost-user-gpu
-#XXX we'll deal with this if some actual need arises,
-#XXX by moving that binary back to q-s-common or packaging it separately
-#Multi-Arch: same
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${misc:Depends}, ${shlibs:Depends},
-# ui-* depends on ui-opengl
- qemu-system-modules-opengl (= ${binary:Version}),
-# we need to ensure qemu-system-gui is upgraded with qemu-system-*
- qemu-system-any (= ${binary:Version}),
-# libgl1 is dynamically loaded by sdl display code
- libgl1,
-# we moved vhost-user-gpu files here from qemu-system-common at 6.1-4
-Replaces: qemu-system-common (<< 1:6.1+dfsg-4~)
-Description: QEMU full system emulation binaries (graphical display and audio modules)
- This package provides optional graphical guest display modules (currently GTK
- and SDL) and audio backend modules for full system emulation (qemu-system-*)
- packages.
- .
- This package is not a management/control/GUI interface for qemu, use something
- else (like virt-manager) for that.
 
-Package: qemu-system-modules-spice
-Architecture: :spice-arch:
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${misc:Depends}, ${shlibs:Depends},
-# spice modules depends on ui-opengl
- qemu-system-modules-opengl (= ${binary:Version}),
-# we need to ensure qemu-system-modules-spice is upgraded with qemu-system-*
- qemu-system-any (= ${binary:Version}),
-Replaces: qemu-system-common (<< 1:8.1.0+ds-1~exp2~)
-Breaks:   qemu-system-common (<< 1:8.1.0+ds-1~exp2~)
-Description: QEMU full system emulation binaries (spice display modules)
- This package provides optional spice display (qxl and spice-app) and audio
- support modules for QEMU full system emulation (qemu-system-*) packages.
 
-Package: qemu-system-modules-opengl
-Architecture: :system-arch:
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${misc:Depends}, ${shlibs:Depends},
-# we need to ensure qemu-system-modules-opengl is upgraded when qemu-system is upgraded
- qemu-system-any (= ${binary:Version}),
-Replaces: qemu-system-common (<< 1:8.1.0+ds-1~exp2~), qemu-system-gui (<< 1:8.1.0+ds-1~exp2~)
-Breaks:   qemu-system-common (<< 1:8.1.0+ds-1~exp2~), qemu-system-gui (<< 1:8.1.0+ds-1~exp2~)
-Description: QEMU full system emulation binaries (OpenGL display modules)
- This package provides optional OpenGL display support modules for QEMU full
- system emulation (qemu-system-*) packages.  It also provides D-Bus display
- type.
 
-Package: qemu-system-misc
-Architecture: :system-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${shlibs:Depends}, ${misc:Depends},
- qemu-system-common (= ${binary:Version}),
- qemu-system-data (>> ${source:Upstream-Version}~),
-# depends on qemu-system-riscv & -s390x since 9.1.1 for now to avoid breaking
-# existing setups which used to get riscv&s390 binaries from -misc.
-# Downgrade to Recommends for one release cycle once tests are fixed
- qemu-system-riscv,
-:debian: qemu-system-s390x,
-Recommends: qemu-utils,
-# alpha uses vgabios
-# alpha m68k sh4 uses bootroms
- seabios,
- ipxe-qemu,
- qemu-system-gui (= ${binary:Version}),
- qemu-system-modules-spice (= ${binary:Version}) [:spice-arch:],
- qemu-system-modules-opengl (= ${binary:Version}),
- qemu-block-extra (= ${binary:Version}),
-Suggests: samba, vde2,
-Provides: ${qemu:Provides}
-Conflicts: ${qemu:Conflicts}
-Description: QEMU full system emulation binaries (miscellaneous)
- QEMU is a fast processor emulator: currently the package supports Alpha, ARM,
- CRIS, i386, LoongArch, M68k (ColdFire), MicroBlaze, MIPS, PowerPC, RISC-V,
- S390x, SH4, SPARC, x86-64, Xtensa and other emulations. By using dynamic
- translation it achieves reasonable speed while being easy to port on new host
- CPUs.
- .
- This package provides the full system emulation binaries to emulate
- various other hardware which did not made into separate packages.
- Emulators for the following architectures are provided:
-  ${qemu:archlist}.
- .
- In system emulation mode QEMU emulates a full system, including a processor
- and various peripherals.  It enables easier testing and debugging of system
- code.  It can also be used to provide virtual hosting of several virtual
- machines on a single server.
 
-Package: qemu-system-arm
-Architecture: :system-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${shlibs:Depends}, ${misc:Depends}, qemu-system-common (= ${binary:Version}), qemu-system-data (>> ${source:Upstream-Version}~),
-Recommends: qemu-utils,
-# aarch64 arm uses bootroms
- ipxe-qemu,
-# for now virtio-vga won't check arch and insist on vgabios-virtio.bin
- seabios,
- qemu-efi-aarch64, qemu-efi-arm,
- qemu-system-gui (= ${binary:Version}),
- qemu-system-modules-spice (= ${binary:Version}) [:spice-arch:],
- qemu-system-modules-opengl (= ${binary:Version}),
- qemu-block-extra (= ${binary:Version}),
-Suggests: samba, vde2,
-Provides: ${qemu:Provides}
-Conflicts: ${qemu:Conflicts}
-Description: QEMU full system emulation binaries (arm)
- QEMU is a fast processor emulator: currently the package supports
- ARM emulation. By using dynamic translation it achieves
- reasonable speed while being easy to port on new host CPUs.
- .
- This package provides the full system emulation binaries to emulate
- the following arm hardware: ${qemu:archlist}.
- .
- In system emulation mode QEMU emulates a full system, including a processor
- and various peripherals.  It enables easier testing and debugging of system
- code.  It can also be used to provide virtual hosting of several virtual
- machines on a single server.
 
-Package: qemu-system-mips
-Architecture: :system-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${shlibs:Depends}, ${misc:Depends}, qemu-system-common (= ${binary:Version}), qemu-system-data (>> ${source:Upstream-Version}~),
-Recommends: qemu-utils,
-# all mips targets uses vgabios and bootroms
- seabios,
- ipxe-qemu,
- qemu-system-gui (= ${binary:Version}),
- qemu-system-modules-spice (= ${binary:Version}) [:spice-arch:],
- qemu-system-modules-opengl (= ${binary:Version}),
- qemu-block-extra (= ${binary:Version}),
-Suggests: samba, vde2,
-Provides: ${qemu:Provides}
-Conflicts: ${qemu:Conflicts}
-Description: QEMU full system emulation binaries (mips)
- QEMU is a fast processor emulator: currently the package supports
- MIPS emulation. By using dynamic translation it achieves
- reasonable speed while being easy to port on new host CPUs.
- .
- This package provides the full system emulation binaries to emulate
- the following mips hardware: ${qemu:archlist}.
- .
- In system emulation mode QEMU emulates a full system, including a processor
- and various peripherals.  It enables easier testing and debugging of system
- code.  It can also be used to provide virtual hosting of several virtual
- machines on a single server.
 
-Package: qemu-system-ppc
-Architecture: :system-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${shlibs:Depends}, ${misc:Depends}, qemu-system-common (= ${binary:Version}), qemu-system-data (>> ${source:Upstream-Version}~),
-Recommends: qemu-utils,
-# ppc targets use vgabios-stdvga and bootroms
- seabios,
- ipxe-qemu,
- qemu-system-gui (= ${binary:Version}),
- qemu-system-modules-spice (= ${binary:Version}) [:spice-arch:],
- qemu-system-modules-opengl (= ${binary:Version}),
- qemu-block-extra (= ${binary:Version}),
-Suggests: samba, vde2,
-Provides: ${qemu:Provides}
-Conflicts: ${qemu:Conflicts}
-Description: QEMU full system emulation binaries (ppc)
- QEMU is a fast processor emulator: currently the package supports
- PowerPC emulation. By using dynamic translation it achieves
- reasonable speed while being easy to port on new host CPUs.
- .
- This package provides the full system emulation binaries to emulate
- the following PowerPC hardware: ${qemu:archlist}.
- .
- In system emulation mode QEMU emulates a full system, including a processor
- and various peripherals.  It enables easier testing and debugging of system
- code.  It can also be used to provide virtual hosting of several virtual
- machines on a single server.
 
-Package: qemu-system-riscv
-Architecture: :system-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${shlibs:Depends}, ${misc:Depends},
- qemu-system-common (= ${binary:Version}),
- qemu-system-data (>> ${source:Upstream-Version}~),
- opensbi (>> 1.5.1-1~),
-Recommends: qemu-utils,
- ipxe-qemu,
-Breaks:   qemu-system-data (<< 1:9.1.0+ds-4~), qemu-system-misc (<< 1:9.1.0+ds-9~)
-Replaces: qemu-system-data (<< 1:9.1.0+ds-4~), qemu-system-misc (<< 1:9.1.0+ds-9~)
-Suggests: samba, vde2,
-Provides: ${qemu:Provides}
-Conflicts: ${qemu:Conflicts}
-Description: QEMU full system emulation binaries (riscv)
- QEMU is a fast processor emulator: currently the package supports
- riscv emulation. By using dynamic translation it achieves
- reasonable speed while being easy to port on new host CPUs.
- .
- This package provides the full system emulation binaries to emulate
- the following riscv hardware: ${qemu:archlist}.
 
-Package: qemu-system-s390x
-Architecture: :system-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${shlibs:Depends}, ${misc:Depends}, qemu-system-common (= ${binary:Version}), qemu-system-data (>> ${source:Upstream-Version}~),
-Recommends: qemu-utils,
- qemu-system-gui (= ${binary:Version}),
- qemu-system-modules-spice (= ${binary:Version}) [:spice-arch:],
- qemu-system-modules-opengl (= ${binary:Version}),
- qemu-block-extra (= ${binary:Version}),
-:debian:Breaks:   qemu-system-misc (<< 1:9.1.0+ds-9~)
-:debian:Replaces: qemu-system-misc (<< 1:9.1.0+ds-9~)
-Provides: ${qemu:Provides}
-Conflicts: ${qemu:Conflicts}
-Description: QEMU full system emulation binaries (s390x)
- QEMU is a fast processor emulator: currently the package supports
- s390x emulation. By using dynamic translation it achieves reasonable
- speed while being easy to port on new host CPUs.
- .
- This package provides the full system emulation binaries to emulate
- the following s390x hardware: ${qemu:archlist}.
- .
- In system emulation mode QEMU emulates a full system, including a processor
- and various peripherals.  It enables easier testing and debugging of system
- code.  It can also be used to provide virtual hosting of several virtual
- machines on a single server.
 
-Package: qemu-system-sparc
-Architecture: :system-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-system>
-Depends: ${shlibs:Depends}, ${misc:Depends}, qemu-system-common (= ${binary:Version}), qemu-system-data (>> ${source:Upstream-Version}~),
-Recommends: qemu-utils,
-# sparc64 uses vgabios-stdvga and bootroms
- seabios,
- ipxe-qemu,
- qemu-system-gui (= ${binary:Version}),
- qemu-system-modules-spice (= ${binary:Version}) [:spice-arch:],
- qemu-system-modules-opengl (= ${binary:Version}),
- qemu-block-extra (= ${binary:Version}),
-Suggests: samba, vde2,
-Provides: ${qemu:Provides}
-Conflicts: ${qemu:Conflicts}
-Description: QEMU full system emulation binaries (sparc)
- QEMU is a fast processor emulator: currently the package supports
- SPARC emulation. By using dynamic translation it achieves
- reasonable speed while being easy to port on new host CPUs.
- .
- This package provides the full system emulation binaries to emulate
- the following sparc hardware: ${qemu:archlist}.
- .
- In system emulation mode QEMU emulates a full system, including a processor
- and various peripherals.  It enables easier testing and debugging of system
- code.  It can also be used to provide virtual hosting of several virtual
- machines on a single server.
 
-:ubuntu:# xen support generally is disabled on ubuntu, for a while we used
-:ubuntu:# an extra build with xen enabled. In the meantime Debian followed that
-:ubuntu:# approach but with a different name, so add a transitional until
-:ubuntu:# after 24.04
-:ubuntu:Package: qemu-system-x86-xen
-:ubuntu:Architecture: amd64
-:ubuntu:Multi-Arch: foreign
-:ubuntu:Section: oldlibs
-:ubuntu:Depends: qemu-system-xen (>= 1:7.0+dfsg-7ubuntu1), ${misc:Depends}
-:ubuntu:Description: QEMU full system emulation binaries (x86)
-:ubuntu: The former qemu-system-x86-xen binaries are now in qemu-system-xen.
-:ubuntu: .
-:ubuntu: This is a transitional package. You can safely remove it.
 
 Package: qemu-system-x86
 Architecture: :system-arch:
@@ -595,9 +324,6 @@ Depends: ${shlibs:Depends}, ${misc:Depends}, qemu-system-common (= ${binary:Vers
  ipxe-qemu,
 Recommends: qemu-utils,
  ovmf,
- qemu-system-gui (= ${binary:Version}),
- qemu-system-modules-spice (= ${binary:Version}) [:spice-arch:],
- qemu-system-modules-opengl (= ${binary:Version}),
  qemu-block-extra (= ${binary:Version}),
 :ubuntu: cpu-checker,
 Suggests: samba, vde2,
@@ -619,94 +345,9 @@ Description: QEMU full system emulation binaries (x86)
  On x86 host hardware this package also enables KVM kernel virtual machine
  usage on systems which supports it.
 
-Package: qemu-system-xen
-Architecture: amd64
-Multi-Arch: no
-Build-Profiles: <!pkg.qemu.omit-system-xen>
-# do we really need qemu-system-data? keymaps only?
-Depends: ${shlibs:Depends}, ${misc:Depends}, qemu-system-data (>> ${source:Upstream-Version}~),
- seabios, ipxe-qemu
-Recommends: qemu-utils,
- ovmf,
-:ubuntu:# For the transition from the former qemu-system-x86-xen name
-:ubuntu:Breaks: qemu-system-x86-xen (<<1:7.0+dfsg-7ubuntu1)
-:ubuntu:Replaces: qemu-system-x86-xen (<<1:7.0+dfsg-7ubuntu1)
-:ubuntu:Provides: qemu-system-x86-xen
-Description: QEMU full system emulation (Xen helper package)
- This package provides the i386 system emulation binary to work
- together with the Xen hypervisor for some types of DomUs.
- This package is not useful by its own.
 
-Package: qemu-user
-Architecture: :user-arch:
-Multi-Arch: foreign
-Built-Using: ${built-using}
-Build-Profiles: <!pkg.qemu.omit-user>
-Depends: ${misc:Depends}
-Breaks:   qemu-user-static (<< 1:9.1.0), qemu-user-binfmt (<< 1:9.1.0)
-Replaces: qemu-user-static (<< 1:9.1.0), qemu-user-binfmt (<< 1:9.1.0)
-Recommends: qemu-user-binfmt
-Description: QEMU user mode emulation (static binaries)
- QEMU is a fast processor emulator: currently the package supports Alpha, ARM,
- CRIS, i386, LoongArch, M68k (ColdFire), MicroBlaze, MIPS, PowerPC, RISC-V,
- S390x, SH4, SPARC, x86-64, Xtensa and other emulations. By using dynamic
- translation it achieves reasonable speed while being easy to port on new host
- CPUs.
- .
- This package provides statically linked user mode emulation binaries.
- In user mode QEMU can launch Linux processes compiled for one CPU
- on another CPU.
- .
- If qemu-user-binfmt package is also installed, it will register binary
- format handlers from this qemu-user package with the kernel automatically
- so it will be possible to run foreign binaries directly.  Without
- qemu-user-binfmt, it is possible to register certain binfmt(s) by using:
-     cat /usr/share/qemu/binfmt.d/qemu-ARCH.conf \
-       > /proc/sys/fs/binfmt_misc/register
 
-# remove this package in favour of qemu-user-binfmt in the future
-Package: qemu-user-static
-Architecture: :user-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-user>
-Depends: ${misc:Depends},
- qemu-user (>= ${binary:Version}), qemu-user-binfmt (>= ${binary:Version})
-Section: oldlibs
-Description: QEMU user mode emulation (compat/transitional package)
- In the past, this package provided statically-built qemu user-mode emulation
- binaries. Now this functionality is provided by qemu-user package, and this
- package just to help migration to qemu-user.
- .
- This package provides old/compatibility symlinks from qemu-user-static binaries
- pointing to corresponding qemu-user binaries, and old qemu-debootstrap script.
- Neither of these should be used in practice.
- .
- Please remove this package once you ensure nothing is using the old interfaces.
 
-Package: qemu-user-binfmt
-Architecture: :user-arch:
-Multi-Arch: foreign
-Build-Profiles: <!pkg.qemu.omit-user>
-Depends: ${misc:Depends}, qemu-user (= ${binary:Version})
-Breaks:   qemu-user-static (<< 1:9.1.0)
-Replaces: qemu-user-static (<< 1:9.1.0)
-# Recommend systemd for binfmt-misc registration
-Recommends: systemd
-# When qemu-user-static is removed, this package should provide it,
-# since the old user-static provided automatic binfmt registration
-#Provides: qemu-user-static
-Description: QEMU user mode binfmt registration for qemu-user
- QEMU is a fast processor emulator: currently the package supports Alpha, ARM,
- CRIS, i386, LoongArch, M68k (ColdFire), MicroBlaze, MIPS, PowerPC, RISC-V,
- S390x, SH4, SPARC, x86-64, Xtensa and other emulations. By using dynamic
- translation it achieves reasonable speed while being easy to port on new host
- CPUs.
- .
- This package provides binfmt support registration for the user-mode emulation
- binaries from qemu-user.  This package does not contain additional binaries,
- just (sym)links to binfmt registration data in /usr/lib/binfmt.d/ so it is
- picked up automatically by, eg, systemd.  Actual files are provided by
- qemu-user package.
 
 Package: qemu-utils
 Architecture: :utils-arch:
diff --git a/debian/control.mk b/debian/control.mk
index da01b80d0e..92757fafe5 100755
--- a/debian/control.mk
+++ b/debian/control.mk
@@ -32,39 +32,8 @@ vdso-files := \
  ${empty}
 
 user-targets := \
- aarch64 \
- aarch64_be \
- alpha \
- arm \
- armeb \
- hexagon \
- hppa \
  i386 \
- loongarch64 \
- m68k \
- microblaze \
- microblazeel \
- mips \
- mips64 \
- mips64el \
- mipsel \
- mipsn32 \
- mipsn32el \
- or1k \
- ppc \
- ppc64 \
- ppc64le \
- riscv32 \
- riscv64 \
- s390x \
- sh4 \
- sh4eb \
- sparc \
- sparc32plus \
- sparc64 \
  x86_64 \
- xtensa \
- xtensaeb \
  ${empty}
 
 # qemu-system (softmmu) targets, in multiple packages
@@ -73,7 +42,7 @@ user-targets := \
 # For each of ${system-archlist-*}, optional:
 #  system-alias-$qcpu   - aliases for this qemu architecture
 
-system-packages := arm mips ppc riscv s390x sparc x86 misc
+system-packages := x86
 
 system-archlist-arm := aarch64 arm
 system-alias-aarch64 := arm64
diff --git a/debian/not-installed b/debian/not-installed
index e3f9930138..12469ae6cd 100644
--- a/debian/not-installed
+++ b/debian/not-installed
@@ -14,3 +14,24 @@ usr/bin/qemu-edid
 usr/share/icons
 usr/share/applications/qemu.desktop
 usr/share/qemu/keymaps
+usr/share/locale/*
+usr/bin/qemu-system-*
+usr/share/doc/qemu/user/*
+usr/lib/*/qemu/audio-dbus.so
+usr/lib/*/qemu/audio-jack.so
+usr/lib/*/qemu/audio-pa.so
+usr/lib/*/qemu/audio-pipewire.so
+usr/lib/*/qemu/audio-sdl.so
+usr/lib/*/qemu/audio-spice.so
+usr/lib/*/qemu/chardev-spice.so
+usr/lib/*/qemu/hw-display-qxl.so
+usr/lib/*/qemu/hw-display-virtio-gpu-gl.so
+usr/lib/*/qemu/hw-display-virtio-gpu-pci-gl.so
+usr/lib/*/qemu/hw-display-virtio-vga-gl.so
+usr/lib/*/qemu/ui-dbus.so
+usr/lib/*/qemu/ui-egl-headless.so
+usr/lib/*/qemu/ui-gtk.so
+usr/lib/*/qemu/ui-opengl.so
+usr/lib/*/qemu/ui-sdl.so
+usr/lib/*/qemu/ui-spice-app.so
+usr/lib/*/qemu/ui-spice-core.so
diff --git a/debian/qemu-system-common.install b/debian/qemu-system-common.install
index ef7f12c481..035c055e01 100644
--- a/debian/qemu-system-common.install
+++ b/debian/qemu-system-common.install
@@ -6,6 +6,8 @@ b/docs/qemu-qmp-ref.7 usr/share/man/man7/
 b/docs/qemu-cpu-models.7 usr/share/man/man7/
 
 usr/share/qemu/trace-events-all
+usr/share/qemu/vhost-user/50-qemu-gpu.json
+usr/lib/qemu/vhost-user-gpu
 usr/bin/qemu-pr-helper
 b/docs/qemu-pr-helper.8 usr/share/man/man8/
 
@@ -15,7 +17,6 @@ usr/lib/qemu/qemu-bridge-helper
 # common modules. Other gui modules are in qemu-system-gui
 usr/lib/${DEB_HOST_MULTIARCH}/qemu/audio-alsa.so
 usr/lib/${DEB_HOST_MULTIARCH}/qemu/audio-oss.so
-usr/lib/${DEB_HOST_MULTIARCH}/qemu/chardev-baum.so
 usr/lib/${DEB_HOST_MULTIARCH}/qemu/hw-display-virtio-gpu-pci.so
 usr/lib/${DEB_HOST_MULTIARCH}/qemu/hw-display-virtio-gpu.so
 usr/lib/${DEB_HOST_MULTIARCH}/qemu/hw-display-virtio-vga.so
diff --git a/debian/rules b/debian/rules
index 078ceaadec..097765661b 100755
--- a/debian/rules
+++ b/debian/rules
@@ -44,7 +44,6 @@ BUILD_PACKAGES := $(shell dh_listpackages)
 enable-system = $(if $(filter qemu-system,${BUILD_PACKAGES}),y)
 enable-user = $(if $(filter qemu-user,${BUILD_PACKAGES}),y)
 
-QEMU_XEN = /usr/libexec/xen-qemu-system-i386
 PKGVERSION = Debian ${DEB_VERSION}
 sysdatadir = /usr/share/qemu
 sysdataidir = debian/qemu-system-data${sysdatadir}
@@ -160,6 +159,7 @@ b/qemu/configured: configure
 		--disable-user --disable-linux-user \
 		--enable-tools \
 		--disable-xen \
+		--target-list=i386-softmmu,x86_64-softmmu \
 		--enable-modules \
 		$(if ${enable-system},--enable-module-upgrades) \
 		$(shell sh debian/extract-config-opts \
@@ -296,37 +296,6 @@ install-microvm: b/microvm/qemu-system-x86_64
 # build microvm on amd64 only if system build is enabled
 qemu-builds += $(if $(filter ${DEB_HOST_ARCH}-${enable-system},amd64-y),microvm)
 
-##############################################
-# xen build (amd64 arch only, i386-softmmu target only)
-configure-xen: b/xen/configured
-b/xen/configured: configure
-	# system build for qemu-system-xen
-	rm -rf b/xen; mkdir -p b/xen
-	cd b/xen && \
-	    ../../configure ${common_configure_opts} \
-		--extra-cflags="${extra-cflags}" \
-		--extra-ldflags="${extra-ldflags}" \
-		--disable-tools \
-		--without-default-features \
-		--enable-avx2 \
-		--enable-xen --target-list=i386-softmmu \
-		--enable-xen-pci-passthrough \
-		--disable-tcg --disable-kvm \
-		--audio-drv-list= \
-		--enable-libusb \
-		--enable-pixman --enable-vnc --enable-vnc-jpeg \
-		--enable-spice \
-		--enable-virtfs --enable-attr --enable-cap-ng \
-		${QEMU_XEN_CONFIGURE_OPTIONS}
-	touch $@
-build-xen: b/xen/built
-b/xen/built: b/xen/configured
-	${NINJA} -C b/xen qemu-system-i386
-	touch $@
-install-xen: b/xen/built
-	install -m0755 -D b/xen/qemu-system-i386 \
-	  debian/qemu-system-xen${QEMU_XEN}
-qemu-builds += $(if $(filter qemu-system-xen,${BUILD_PACKAGES}),xen)
 
 ##############################################
 # user build (static)
@@ -473,9 +442,7 @@ install-arch: pre-install-arch $(addprefix install-, ${qemu-builds})
 	dh_fixperms -a
 	dh_dwz -a
 	dh_strip -a
-# qemu-user contains static binaries only, but they might be built
-# as static-pie executables and dpkg-shlibdeps complains. Just omit it here.
-	dh_shlibdeps -a -Nqemu-user
+	dh_shlibdeps -a
 binary-arch: install-arch binary-helper
 binary-arch: a=-a
 
@@ -547,85 +514,17 @@ install-x86-optionrom: b/data/pc-bios/optionrom/.built
 	install -m0644 -t ${sysdataidir} ${<D}/*.bin
 sysdata-components += x86-optionrom
 
-### s390x firmware in pc-bios/s390-ccw
-build-s390x-fw: b/data/pc-bios/s390-ccw/.built
-b/data/pc-bios/s390-ccw/.built: b/data/.configured
-	env -u ELF_PACKAGE_METADATA \
-	make -C ${@D} V=$V
-	touch $@
-install-s390x-fw: b/data/pc-bios/s390-ccw/.built
-	install -m0644 -t ${sysdataidir} ${<D}/s390*.img
-sysdata-components += s390x-fw
+### s390x firmware - removed for STX (x86 only)
 
-### vof.bin
-build-vof: b/data/pc-bios/vof/vof.bin
-b/data/pc-bios/vof/vof.bin: b/data/.configured
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} -C ${@D} V=$V ${@F}
-install-vof: b/data/pc-bios/vof/vof.bin
-	install -m0644 -t ${sysdataidir} $<
-sysdata-components += vof
-
-### openbios rules
-b/openbios/config-host.mak:
-	mkdir -p b/openbios
-	cd b/openbios && ../../roms/openbios/config/scripts/switch-arch builtin-ppc builtin-sparc32 builtin-sparc64
-build-openbios: $(patsubst %,b/openbios/obj-%/.built, ppc sparc32 sparc64)
-b/openbios/obj-%/.built: b/openbios/config-host.mak
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} -C ${@D} V=${V} EXTRACFLAGS="-ffreestanding -fno-pic -fno-stack-protector"
-	@touch $@
-
-install-openbios: build-openbios
-	install -m0644 b/openbios/obj-ppc/openbios-qemu.elf ${sysdataidir}/openbios-ppc
-	install -m0644 b/openbios/obj-sparc32/openbios-builtin.elf ${sysdataidir}/openbios-sparc32
-	install -m0644 b/openbios/obj-sparc64/openbios-builtin.elf ${sysdataidir}/openbios-sparc64
-	install -m0644 -t ${sysdataidir} \
-		b/openbios/obj-sparc32/QEMU,tcx.bin \
-		b/openbios/obj-sparc32/QEMU,cgthree.bin \
-		b/openbios/obj-sparc64/QEMU,VGA.bin
-sysdata-components += openbios
-
-### powernv firmware in roms/skiboot
-build-skiboot: b/skiboot/skiboot.lid
-b/skiboot/skiboot.lid: | roms/skiboot/.version
-	mkdir -p b/skiboot
-# skiboot makefiles makes it difficult to *add* an option to CFLAGS.
-# Abuse OPTS= for this, with the default being -Os.
-	grep -q '^OPTS=-Os$$' roms/skiboot/Makefile.main || \
-	  { echo "review OPTS= in skiboot/Makefile.main"; false; }
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} -C b/skiboot -f ${CURDIR}/roms/skiboot/Makefile \
-	  SRC=${CURDIR}/roms/skiboot \
-	  OPTS='-Os -ffile-prefix-map="${CURDIR}/roms/skiboot/"=' \
-	  CROSS_COMPILE=${cross_prefix_ppc64} V=${V}
-install-skiboot: b/skiboot/skiboot.lid
-	install -m0644 -t ${sysdataidir} $<
-sysdata-components += skiboot
+### vof.bin - removed for STX (x86 only)
 
-### u-boot-e500 (u-boot.e500)
-build-u-boot-e500: b/u-boot/build-e500/u-boot
-b/u-boot/build-e500/u-boot: | b
-	cp -alu roms/u-boot b/
-	${MAKE} -C b/u-boot CROSS_COMPILE=${cross_prefix_ppc} O=build-e500 qemu-ppce500_config
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} -C b/u-boot CROSS_COMPILE=${cross_prefix_ppc} O=build-e500 V=$V
-	${cross_prefix_ppc}strip $@
-install-u-boot-e500: b/u-boot/build-e500/u-boot
-	install -m0644 $< ${sysdataidir}/u-boot.e500
-sysdata-components += u-boot-e500
-
-### u-boot-sam460 (u-boot-sam460-20100605.bin)
-build-u-boot-sam460: b/u-boot-sam460ex/u-boot.bin
-b/u-boot-sam460ex/u-boot.bin: | b
-	cp -alu roms/u-boot-sam460ex b/
-	${MAKE} -C b/u-boot-sam460ex CROSS_COMPILE=${cross_prefix_ppc} Sam460ex_config
-	env -u LDFLAGS -u OBJCFLAGS -u ELF_PACKAGE_METADATA \
-	${MAKE} -C b/u-boot-sam460ex CROSS_COMPILE=${cross_prefix_ppc}
-#	${cross_prefix_ppc}strip $@
-install-u-boot-sam460: b/u-boot-sam460ex/u-boot.bin | ${sysdataidir}
-	install -m0644 $< ${sysdataidir}/u-boot-sam460-20100605.bin
-sysdata-components += u-boot-sam460
+### openbios rules - removed for STX (x86 only)
+
+### powernv firmware in roms/skiboot - removed for STX (x86 only)
+
+### u-boot-e500 - removed for STX (x86 only)
+
+### u-boot-sam460 - removed for STX (x86 only)
 
 ### qboot, aka bios-microvm
 build-qboot: b/qboot/bios.bin
@@ -638,62 +537,13 @@ install-qboot: b/qboot/bios.bin
 	install -m0644 $< ${sysdataidir}/qboot.rom
 sysdata-components += qboot
 
-### alpha firmware in roms/palcode-clipper
-build-palcode-clipper: b/qemu-palcode/palcode-clipper
-b/qemu-palcode/palcode-clipper: | b
-	cp -al roms/qemu-palcode b/
-#XXX #1019011 (remove OPT= alternative when fixed)
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} -C b/qemu-palcode CROSS=${cross_prefix_alpha} -k || \
-	${MAKE} -C b/qemu-palcode CROSS=${cross_prefix_alpha} OPT=-O1
-	${cross_prefix_alpha}strip b/qemu-palcode/palcode-clipper
-install-palcode-clipper: b/qemu-palcode/palcode-clipper
-	install -m0644 $< ${sysdataidir}/palcode-clipper
-sysdata-components += palcode-clipper
-
-### SLOF
-build-slof: b/SLOF/boot_rom.bin
-b/SLOF/boot_rom.bin: | b
-	cp -al roms/SLOF b/
-	env -u LDFLAGS -u CFLAGS -u ELF_PACKAGE_METADATA \
-	$(MAKE) -C b/SLOF qemu CROSS=${cross_prefix_ppc64} V=${V}
-install-slof: b/SLOF/boot_rom.bin
-	install -m0644 $< ${sysdataidir}/slof.bin
-sysdata-components += slof
-
-### hppa-firmware (roms/seabios-hppa)
-build-hppa-fw: b/hppa-fw/.built
-b/hppa-fw/.built: | b
-	rm -rf $(dir $@)
-	cp -al roms/seabios-hppa $(dir $@)
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} PYTHON=python3 OUT=o32/ BITS=32 BIT_SUFFIX=   CROSS_PREFIX=${cross_prefix_hppa}   -C $(dir $@) -f Makefile.parisc all
-	${cross_prefix_hppa}strip   -R.note -R.comment $(dir $@)o32/hppa-firmware.img
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} PYTHON=python3 OUT=o64/ BITS=64 BIT_SUFFIX=64 CROSS_PREFIX=${cross_prefix_hppa64} -C $(dir $@) -f Makefile.parisc all
-	${cross_prefix_hppa64}strip -R.note -R.comment $(dir $@)o64/hppa-firmware64.img
-	touch $@
-install-hppa-fw: b/hppa-fw/.built
-	install -m0644 -t ${sysdataidir} \
-	    $(dir $<)o32/hppa-firmware.img \
-	    $(dir $<)o64/hppa-firmware64.img
-sysdata-components += hppa-fw
-
-### vbootrom (npcm7xx, npcm8xx)
-build-vbootrom: b/vbootrom/.built
-b/vbootrom/.built: | b
-	cp -pa roms/vbootrom b/
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} -C b/vbootrom/npcm7xx CROSS_COMPILE=${cross_prefix_arm}
-	env -u ELF_PACKAGE_METADATA \
-	${MAKE} -C b/vbootrom/npcm8xx CROSS_COMPILE=${cross_prefix_aarch64}
-	touch $@
-install-vbootrom: build-vbootrom
-	install -m0644 \
-	   b/vbootrom/npcm7xx/npcm7xx_bootrom.bin \
-	   b/vbootrom/npcm8xx/npcm8xx_bootrom.bin \
-		-t ${sysdataidir}/
-sysdata-components += vbootrom
+### alpha firmware - removed for STX (x86 only)
+
+### SLOF - removed for STX (x86 only)
+
+### hppa-firmware - removed for STX (x86 only)
+
+### vbootrom - removed for STX (x86 only)
 
 ### misc firmware
 build-misc: b/misc/.built
-- 
2.34.1

